File: javascript/test/crc.js
#!/usr/bin/env node
var liba = require("../liba");
function assert(expr, info) {
    if (!expr) {
        throw Error("assertion failed: " + info);
    }
}
function do_leak_check() {
    for (const arg of arguments) {
        arg.delete();
    }
    if (liba.do_leak_check) {
        liba.do_leak_check();
    }
}
var text = "0123456789";
var crc = new liba.crc8(0x07);
assert(crc.eval(text, 0) == 0x45);
assert(crc.eval(crc.pack(text, 0), 0) == 0);
do_leak_check(crc);
var crc = new liba.crc8(0x31, true);
assert(crc.eval(text, 0) == 0x75);
assert(crc.eval(crc.pack(text, 0), 0) == 0);
do_leak_check(crc);
var crc = new liba.crc16(0x1021);
assert(crc.eval(text, 0) == 0x9C58);
assert(crc.eval(crc.pack(text, 0), 0) == 0);
do_leak_check(crc);
var crc = new liba.crc16(0x8005, true);
assert(crc.eval(text, 0) == 0x443D);
assert(crc.eval(crc.pack(text, 0), 0) == 0);
do_leak_check(crc);
var crc = new liba.crc32(0x1EDC6F41);
assert(crc.eval(text, 0) == 0x512B456E);
assert(crc.eval(crc.pack(text, 0), 0) == 0);
do_leak_check(crc);
var crc = new liba.crc32(0x04C11DB7, true);
assert(crc.eval(text, 0) == 0x450EAFB0);
assert(crc.eval(crc.pack(text, 0), 0) == 0);
do_leak_check(crc);
if (liba.crc64) {
    var crc = new liba.crc64(0x000000000000001Bn);
    assert(crc.eval(text, 0n) == 0xE4FFBEA588AFC790n);
    assert(crc.eval(crc.pack(text, 0n), 0n) == 0n);
    do_leak_check(crc);
    var crc = new liba.crc64(0x42F0E1EBA9EA3693n, true);
    assert(crc.eval(text, 0n) == 0xDA60676A5CDE0008n);
    assert(crc.eval(crc.pack(text, 0n), 0n) == 0n);
    do_leak_check(crc);
}

File: java/test/_crc.java
import java.util.Arrays;
import liba.crc8;
import liba.crc16;
import liba.crc32;
import liba.crc64;

class _crc {
    public static void main(String[] args) {
        String text = "0123456789";
        {
            crc8 obj = new crc8((byte) 0x07, false);
            assert obj.eval(text.getBytes()) == (byte) 0x45;
            assert obj.eval(obj.pack(text.getBytes())) == 0;
            System.out.println(Arrays.toString(obj.table()));
        }
        {
            crc8 obj = new crc8((byte) 0x31, true);
            assert obj.eval(text.getBytes()) == (byte) 0x75;
            assert obj.eval(obj.pack(text.getBytes())) == 0;
            System.out.println(Arrays.toString(obj.table()));
        }
        {
            crc16 obj = new crc16((short) 0x1021, false);
            assert obj.eval(text.getBytes()) == (short) 0x9C58;
            assert obj.eval(obj.pack(text.getBytes())) == 0;
            System.out.println(Arrays.toString(obj.table()));
        }
        {
            crc16 obj = new crc16((short) 0x8005, true);
            assert obj.eval(text.getBytes()) == (short) 0x443D;
            assert obj.eval(obj.pack(text.getBytes())) == 0;
            System.out.println(Arrays.toString(obj.table()));
        }
        {
            crc32 obj = new crc32(0x1EDC6F41, false);
            assert obj.eval(text.getBytes()) == 0x512B456E;
            assert obj.eval(obj.pack(text.getBytes())) == 0;
            System.out.println(Arrays.toString(obj.table()));
        }
        {
            crc32 obj = new crc32(0x04C11DB7, true);
            assert obj.eval(text.getBytes()) == 0x450EAFB0;
            assert obj.eval(obj.pack(text.getBytes())) == 0;
            System.out.println(Arrays.toString(obj.table()));
        }
        {
            crc64 obj = new crc64(0x000000000000001BL, false);
            assert obj.eval(text.getBytes()) == 0xE4FFBEA588AFC790L;
            assert obj.eval(obj.pack(text.getBytes())) == 0;
            System.out.println(Arrays.toString(obj.table()));
        }
        {
            crc64 obj = new crc64(0x42F0E1EBA9EA3693L, true);
            assert obj.eval(text.getBytes()) == 0xDA60676A5CDE0008L;
            assert obj.eval(obj.pack(text.getBytes())) == 0;
            System.out.println(Arrays.toString(obj.table()));
        }
    }
}
